Objects/listobject.c

```diff
@@ -937,7 +937,7 @@ list_ass_slice_lock_held(PyListObject *a, Py_ssize_t ilow, Py_ssize_t ihigh, PyO
     }
     for (k = 0; k < n; k++, ilow++) {
         PyObject *w = vitem[k];
-        item[ilow] = Py_XNewRef(w);
+        FT_ATOMIC_STORE_PTR_RELAXED(item[ilow], Py_XNewRef(w));
```

Comment:
I think this should be `FT_ATOMIC_STORE_PTR_RELEASE`. We generally need `release` when writing pointers that may be load and dereferenced outside of the lock to ensure that the previously written contents are visible before the pointer itself.